relay-server/src/actors/envelopes.rs

```diff
@@ -618,10 +658,6 @@ impl Handler<HandleEnvelope> for EnvelopeManager {
                     relay_log::debug!("dropped envelope: {}", LogError(&error));
                 }
             })
-            .then(move |x, slf, _| {
-                slf.active_envelopes -= 1;
```

Comment:
This is the important bit of this entire change. We no longer depend on the global future to track the number of active envelopes.